reciprocalspaceship/decorators.py

```diff
@@ -46,8 +48,12 @@ def wrapped(ds, *args, **kwargs):
         names = ds.index.names
         ds = ds._index_from_names([None], inplace=True)
         result = f(ds, *args, **kwargs)
-        result = result._index_from_names(names, inplace=True)
         ds = ds._index_from_names(names, inplace=True)
+        if result is None:
```

Comment:
I would prefer to use the following pattern, while acknowledging it is a purely aesthetic preference. 

```python
if isinstance(result, rs.DataSet):
    result = result._index_from_names(names, inplace=True)
    result =  result.__finalize__(ds)
return result
```

Reply:
agreed. Cleaner. I'll update.